code/modules/clothing/suits/ablativecoat.dm

```diff
@@ -47,3 +33,22 @@
 		return FALSE
 	if (prob(hit_reflect_chance))
 		return TRUE
+
+/obj/item/clothing/suit/hooded/ablative/ToggleHood()
+	. = ..()
+	var/mob/living/carbon/user = loc
+	if (hood_up)
+		var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+		to_chat(user, "As you put on the hood, a visor shifts into place and starts analyzing the people around you. Neat!")
+		ADD_TRAIT(user, TRAIT_SECURITY_HUD, HELMET_TRAIT)
+		hud.add_hud_to(user)
+	else
+		to_chat(user, "You take off the hood, removing the visor in the process and disabling its integrated hud.") // send chat message here, since RemoveHood() is called a couple times
```

Comment:
Bad fix, just send this chat message if they have `TRAIT_SECURITY_HUD` in `RemoveHood`.